torchmetrics/functional/text/rouge.py

```diff
@@ -0,0 +1,152 @@
+
+
+def format_rouge_results(result: Dict[str, AggregateScore], decimal_places: int = 4) -> Dict[str, Tensor]:
+    flattened_result = {}
+    for rouge_key, rouge_aggregate_score in result.items():
+        for stat in ["precision", "recall", "fmeasure"]:
+            mid = rouge_aggregate_score.mid
+            score = round(getattr(mid, stat), decimal_places)
+            flattened_result[f"{rouge_key}_{stat}"] = tensor(score)
+    return flattened_result
+
+
+class RougeBatchAggregator(BootstrapAggregator):
+    """
+    Aggregates rouge scores and provides confidence intervals.
+    """
+
+    def aggregate(self) -> Dict[str, AggregateScore]:
+        """
+        Override function to wrap the final results in `Score` objects.
+        This is due to the scores being replaced with a list of torch tensors.
+        """
+        result = {}
+        for score_type, scores in self._scores.items():
+            # Stack scores into a 2-d matrix of (sample, measure).
+            score_matrix = np.vstack(tuple(scores))
+            # Percentiles are returned as (interval, measure).
+            percentiles = self._bootstrap_resample(score_matrix)
+            # Extract the three intervals (low, mid, high).
+            intervals = tuple((Score(*percentiles[j, :]) for j in range(3)))
+            result[score_type] = AggregateScore(low=intervals[0], mid=intervals[1], high=intervals[2])
+        return result
+
+    def add_scores(self, scores) -> None:
+        self._scores = scores
+
+
+def _rouge_score_update(
+    pred_lns: List[str],
+    tgt_lns: List[str],
+    scores: Dict[str, List[Tensor]],
+    scorer: rouge_scorer.RougeScorer,
+    rouge_newline_sep: bool = False,
+) -> None:
+
+    for pred, tgt in zip(pred_lns, tgt_lns):
+        # rougeLsum expects "\n" separated sentences within a summary
+        if rouge_newline_sep:
+            pred = add_newline_to_end_of_each_sentence(pred)
+            tgt = add_newline_to_end_of_each_sentence(tgt)
+        results = scorer.score(pred, tgt)
+        for key, score in results.items():
+            score = tensor([score.precision, score.recall, score.fmeasure])
+            scores[key].append(score)
+
+
+def _rouge_score_compute(scores: Dict[str, List[Tensor]], aggregator: RougeBatchAggregator) -> Dict[str, Tensor]:
+    aggregator.add_scores(scores)
+    result = aggregator.aggregate()
+    return format_rouge_results(result)
+
+
+def rouge_score(
+    pred_lns: List[str],
+    tgt_lns: List[str],
+    rouge_newline_sep: bool = False,
+    use_stemmer: bool = False,
+    rouge_keys: Tuple[str] = ("rouge1", "rouge2", "rougeL", "rougeLsum")
+) -> Dict[str, Tensor]:
+    """
+    Calculate `ROUGE score <https://en.wikipedia.org/wiki/ROUGE_(metric)>`_, used for automatic summarization.
+
+    Args:
+        pred_lns:
+            An iterable of predicted tokens.
+        tgt_lns:
+            An iterable of target tokens.
+        rouge_newline_sep:
+
+        use_stemmer:
```

Comment:
for these two you can copy some of the text from here:
https://github.com/google-research/google-research/blob/7b9d963a56f58ae7e036243e6a5b75d3294175ea/rouge/rouge_scorer.py#L66-L68